azurerm/resource_arm_policy_definition_test.go

```diff
@@ -35,6 +36,64 @@ func TestAccAzureRMPolicyDefinition_basic(t *testing.T) {
 	})
 }
 
+func TestAccAzureRMPolicyDefinitionAtMgmtGroup_basic(t *testing.T) {
+	resourceName := "azurerm_policy_definition.test"
+	mgmtGroupName := "azurerm_management_group.test"
+
+	ri := acctest.RandInt()
+
+	resource.ParallelTest(t, resource.TestCase{
+		PreCheck:     func() { testAccPreCheck(t) },
+		Providers:    testAccProviders,
+		CheckDestroy: testCheckAzureRMPolicyDefinitionDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config: testAzureRMPolicyDefinition_basic(ri, true),
+				Check: resource.ComposeTestCheckFunc(
+					testCheckAzureRMPolicyDefinitionExistsInMgmtGroup(resourceName, mgmtGroupName),
+				),
+			},
+			{
+				ResourceName:      resourceName,
+				ImportState:       true,
+				ImportStateVerify: true,
+			},
+		},
+	})
+}
+
+func testCheckAzureRMPolicyDefinitionExistsInMgmtGroup(policyName string, managementGroupName string) resource.TestCheckFunc {
+	return func(s *terraform.State) error {
+		rs, ok := s.RootModule().Resources[policyName]
+		if !ok {
+			return fmt.Errorf("not found: %s", policyName)
+		}
+
+		policyName := rs.Primary.Attributes["name"]
+
+		rs, ok = s.RootModule().Resources[managementGroupName]
+		if !ok {
+			return fmt.Errorf("not found: %s", managementGroupName)
+		}
```

Comment:
since the `management_group_id` key is available on the resource, we should be able to use this instead of querying the (separate) management group resource - as such I don't think this is needed?

Reply:
yes, this noticeably simplified the test. thanks